lib/stdlib/test/property_test/base64_prop.erl

```diff
@@ -0,0 +1,429 @@
+                false
+        catch
+            error:_ ->
+                true
+        end
+    ).
+
+%%%%%%%%%%%%%%%%%%
+%%% Generators %%%
+%%%%%%%%%%%%%%%%%%
+
+%% Generate a single character from the base64 alphabet.
+b64_char() ->
+    oneof(b64_chars()).
+
+%% Generate a string of characters from the base64 alphabet,
+%% including padding if needed.
+b64_string() ->
+    ?LET(
+        {L, Filler},
+        {list(b64_char()), b64_char()},
+        case length(L) rem 4 of
+            0 -> L;
+            1 -> L ++ [Filler, $=, $=];
+            2 -> L ++ [$=, $=];
+            3 -> L ++ [$=]
+        end
+    ).
+
+%% Generate a binary of characters from the base64 alphabet,
+%% including padding if needed.
+b64_binary() ->
+    ?LET(
+        L,
+        b64_string(),
+        list_to_binary(L)
+    ).
+
+%% Generate a string or binary of characters from the
+%% base64 alphabet, including padding if needed.
+b64() ->
+    oneof([b64_string(), b64_binary()]).
+
+%% Generate a string or binary of characters from the
+%% base64 alphabet, including padding if needed, with
+%% whitespaces inserted at random indexes.
+wsped_b64() ->
+    ?LET(
+        {B64, Wsps},
+        {b64(), list(oneof([$\t, $\r, $\n, $\s]))},
+        {B64, insert_noise(B64, Wsps)}
+    ).
+
+%% Generate a single character outside of the base64 alphabet.
+%% As whitespaces are allowed but ignored in base64, this generator
+%% will produce no whitespaces, either.
+non_b64_char() ->
+    oneof(lists:seq(16#00, 16#FF) -- b64_allowed_chars()).
+
+%% Generate a string or binary of characters from the
+%% base64 alphabet, including padding if needed, with
+%% whitespaces and non-base64 ("invalid") characters
+%% inserted at random indexes.
+noisy_b64() ->
+    ?LET(
+        {{B64, WspedB64}, Noise},
+        {wsped_b64(), non_empty(list(non_b64_char()))},
+        {B64, insert_noise(WspedB64, Noise)}
+    ).
+
+%%%%%%%%%%%%%%%
+%%% Helpers %%%
+%%%%%%%%%%%%%%%
+
+%% The characters of the base64 alphabet.
+%% "=" is not included, as it is special in that it
+%% may only appear at the end of a base64 encoded string
+%% for padding.
+b64_chars() ->
+    lists:seq($0, $9) ++
+    lists:seq($a, $z) ++
+    lists:seq($A, $Z) ++
+    [$+, $/].
+
+%% In addition to the above, the whitespace characters
+%% HTAB, CR, LF and SP are allowed to appear in a base64
+%% encoded string and should be ignored.
+b64_allowed_chars() ->
+    [$\t, $\r, $\n, $\s | b64_chars()].
+
+%% Insert the given list of noise characters at random
+%% places into the given base64 string.
+insert_noise(B64, []) ->
+    B64;
+insert_noise([], Noise) ->
+    Noise;
+insert_noise(<<>>, Noise) ->
+    list_to_binary(Noise);
+insert_noise([B|Bs] = B64, [N|Ns] = Noise) ->
+    case rand:uniform(2) of
+        1 ->
+            [B|insert_noise(Bs, Noise)];
+        2 ->
+            [N|insert_noise(B64, Ns)]
+    end;
+insert_noise(<<B, Bs/binary>> = B64, [N|Ns] = Noise) ->
+    case rand:uniform(2) of
```

Comment:
it is usually a bad idea to use any side-effects in PBT generators, because indeterminism would completely mess shrinking. If you need some source of entropy, it's better to outsource it to the PBT framework (in this case, by using smth like `list(integer(1, 2))`), so the randomness of the generator is fully controlled by the framework.